File: app-back/src/models/accommodation.ts
import { IAccommodationType } from "../interfaces/IAccommodationType";
import { ICity } from "../interfaces/ICity";

export class Accommodation {
  constructor(
    id: number,
    title: string,
    accommodationTypeId: number,
    cityId: number,
    address: string,
    bedroomsCount: string,
    bedsCount: string,
    capacity: string,
    imageName: string,
    accommodationType?: IAccommodationType,
    city?: ICity
  ) {}
}

File: app-api/Model/Accommodation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Net;

namespace app_api.Model
{
    public class Accommodation
    {
        [Key]
        public int Id { get; set; }
        public required string Title { get; set; }
        public  string? Address { get; set; }
        public  string? BedroomsCount { get; set; }
        public  string? BedsCount { get; set; }
        public  string? Capacity { get; set; }
        public  string? ImageName { get; set; }
        [ForeignKey("City")]
        public int CityId { get; set; }
        public required City City { get; set; }
        [ForeignKey("AccommodationType")]
        public int AccommodationTypeId { get; set; }
        public required  AccommodationType AccommodationType { get; set;}
    }
}
